concert/app/src/functions/prepare_stack/index.py

```diff
@@ -0,0 +1,95 @@
+import boto3
+import json
+import os
+from botocore.vendored import requests
+from enchanted_brain.attributes import (
+    ATTR_CHOICE_VALUE_EMOTION,
+    ATTR_CHOICE_VALUE_CHILLS,
+    ATTR_CHOICE_VALUE_COLOR,
+    ATTR_RECORD_ID,
+    RECORD_ID_AGGREGATE,
+)
+
+DYNAMODB_TABLE_NAME = os.environ.get("DYNAMODB_TABLE_NAME")
+DYNAMODB_ENDPOINT = os.environ.get("DYNAMODB_ENDPOINT")
+KINESIS_ANALYTICS_APPLICATION_NAME = os.environ.get(
+    "KINESIS_ANALYTICS_APPLICATION_NAME"
+)
+
+dynamodb_args = {}
+if DYNAMODB_ENDPOINT:
+    dynamodb_args["endpoint_url"] = DYNAMODB_ENDPOINT
+dynamodb = boto3.resource("dynamodb", **dynamodb_args)
+table = dynamodb.Table(DYNAMODB_TABLE_NAME)
+
+kinesis_analytics = boto3.client("kinesisanalyticsv2")
+
+
+def handler(event, context):
+    status = "FAILED"
+    try:
+        prepare_aggregate_choices_record()
+        start_kinesis_analytics_application()
+        status = "SUCCESS"
+    finally:
```

Comment:
It looks like we're swallowing error messages here. Won't the try/finally stop the function from erroring out (which is good), meaning that the error's message won't actually get logged to CloudWatch (which is bad)?

Reply:
If I'm understanding [Python's try/finally behavior](https://docs.python.org/3/tutorial/errors.html#defining-clean-up-actions) correctly, the error will still be raised after completion of the `finally` block, so it should still get logged to CloudWatch

Reply:
Ah, I was wrong about how the try/finally works. But that means this function can error out instead of reporting failure. Is that what we want?

Reply:
IMO it's fine. As far as I can tell, CloudFormation doesn't know or care how the Lambda terminates or what it returns. Termination status is reported instead through the S3 upload that happens in `put_response` within the `finally` block.

(From [AWS's documentation on custom resources](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/template-custom-resources.html): "Custom resources must send responses to a pre-signed Amazon S3 URL. If they can't send responses to Amazon S3, AWS CloudFormation won't receive a response and the stack operation fails.")

Reply:
But that's exactly my point. Instead of a timely and meaningful failure reason reported to the S3 url, cloudformation will just timeout waiting for a response that never comes.

Reply:
Something as simple as this would handle it:
```suggestion
    except Exception as e:
        print("ERROR", e.message)
    put_response(event, context, status)
```

Reply:
Once again, I was overthinking things. The finally block takes care of reporting back to CloudFormation, so erroring out after that is fine.